src/webgpu/shader/execution/expression/call/builtin/ldexp.spec.ts

```diff
@@ -29,9 +29,21 @@ g.test('abstract_float')
 `
   )
   .params(u =>
-    u.combine('inputSource', allInputSources).combine('vectorize', [undefined, 2, 3, 4] as const)
+    u
+      .combine('inputSource', onlyConstInputSource)
+      .combine('vectorize', [undefined, 2, 3, 4] as const)
   )
-  .unimplemented();
+  .fn(async t => {
+    const cases = await d.get('abstract_const');
```

Comment:
Just so I understand, this `abstract_const` will be returning a case which has something like `(0.1, 1)` -> `5.5` or whatever the real values are? Like, it's specific to `ldexp`?

Reply:
Yeah, it will be pulling the 'abstract_const' entry from the CaseCache, d, which is specific to ldexp. The cache side of this test was already written, I just apparently didn't stage the change to the test to actually use it when originally implementing.

Reply:
The name 'abstract_const' comes from how the tests are generated in the .cache.ts, since it effectively generates the cartesian product of keys into the FPTraits and const or non-const. `abstract_non_const` is an empty array.